roles/container-engine/docker/files/cleanup-docker-orphans.sh

```diff
@@ -25,16 +20,20 @@ if [ ${count_shim_processes} -gt 0 ]; then
 
         if [[ -n "$orphans" ]]
         then
-                # Get shim pids of orphans
-                orphan_shim_pids=$(ps -o pid= $(ps -o ppid= $orphans))
-
-                # Find all orphaned container PIDs
-                orphan_container_pids=$(for pid in $orphan_shim_pids; do list_descendants $pid; done)
-
-                # Recursively kill all child PIDs of orphan shims
-                echo -e "Killing orphan container PIDs and descendants: \n$(ps -O ppid= $orphan_container_pids)"
-                kill -9 $orphan_container_pids || true
+                orphan_ppid=$(ps -o ppid= $orphans)
+                if [[ -n "$orphan_ppid" ]]
```

Comment:
Could this be a second condition in the above if statement? Changed to something like `if [[ -n "$orphans" && -n "$(ps -o ppid= $orphans)" ]]` It would read a bit cleaner instead of second nesting another if and else

Reply:
Sure, that would certainly be a cleaner way to do it.